cmd/signozschemamigrator/schema_migrator/logs_migrations_v2.go

```diff
@@ -0,0 +1,441 @@
+					Table:       constants.LocalPathTypesTable,
+					ShardingKey: fmt.Sprintf("cityHash64(%s, %s)", constants.PathTypesTablePathColumn, constants.PathTypesTableTypeColumn),
+				},
+			},
+			CreateTableOperation{
+				Database: SignozMetadataDB,
+				Table:    constants.LocalPromotedPathsTable,
+				Columns: []Column{
+					{Name: "path", Type: ColumnTypeString, Codec: "ZSTD(1)"},
+					{Name: "created_at", Type: ColumnTypeUInt64, Codec: "DoubleDelta, LZ4"},
+				},
+				Engine: ReplacingMergeTree{
+					MergeTree: MergeTree{
+						OrderBy:     "path",
+						PartitionBy: "toDate(created_at / 1000000000)",
+						Settings: TableSettings{
+							{Name: "index_granularity", Value: "8192"},
+						},
+					},
+				},
+			},
+			CreateTableOperation{
+				Database: SignozMetadataDB,
+				Table:    constants.DistributedPromotedPathsTable,
+				Columns: []Column{
+					{Name: "path", Type: ColumnTypeString, Codec: "ZSTD(1)"},
+					{Name: "created_at", Type: ColumnTypeUInt64, Codec: "DoubleDelta, LZ4"},
+				},
+				Engine: Distributed{
+					Database:    SignozMetadataDB,
+					Table:       constants.LocalPromotedPathsTable,
+					ShardingKey: "cityHash64(path)",
+				},
+			},
+			AlterTableModifySettings{
+				Database: "signoz_logs",
+				Table:    "logs_v2",
+				Settings: TableSettings{
+					{Name: "object_serialization_version", Value: "'v3'"},
+					{Name: "object_shared_data_serialization_version", Value: "'advanced'"},
+				},
+			},
+			AlterTableAddColumn{
+				Database: "signoz_logs",
+				Table:    "logs_v2",
+				Column: Column{
+					Name: constants.BodyJSONColumn,
+					Type: JSONColumnType{
+						MaxDynamicPaths: utils.ToPointer[uint](0),
+					},
+					Codec: "ZSTD(1)",
+				},
+				After: &Column{
+					Name: "body",
+				},
+			},
+			AlterTableAddColumn{
+				Database: "signoz_logs",
+				Table:    "distributed_logs_v2",
+				Column: Column{
+					Name: constants.BodyJSONColumn,
+					Type: JSONColumnType{
+						MaxDynamicPaths: utils.ToPointer[uint](0),
+					},
+					Codec: "ZSTD(1)",
+				},
+				After: &Column{
+					Name: "body",
+				},
+			},
+			AlterTableAddColumn{
+				Database: "signoz_logs",
+				Table:    "logs_v2",
+				Column: Column{
+					Name:  constants.BodyPromotedColumn,
+					Type:  JSONColumnType{},
+					Codec: "ZSTD(1)",
+				},
+				After: &Column{
+					Name: constants.BodyJSONColumn,
+				},
+			},
+			AlterTableAddColumn{
+				Database: "signoz_logs",
+				Table:    "distributed_logs_v2",
+				Column: Column{
+					Name:  constants.BodyPromotedColumn,
+					Type:  JSONColumnType{},
+					Codec: "ZSTD(1)",
+				},
+				After: &Column{
+					Name: constants.BodyJSONColumn,
+				},
+			},
+			InsertIntoTable{
+				Database:    SignozMetadataDB,
+				Table:       constants.DistributedPromotedPathsTable,
+				LightWeight: true,
+				Synchronous: true,
+				Columns:     []string{"path", "created_at"},
+				Values:      fmt.Sprintf("('message', %d)", time.Now().UnixMilli()),
```

Comment:
time.now() will not be idempotent i.e you won't known when it was added if this migration is run again which is `INSERT INTO %s.%s (path, created_at) VALUES ('message', %d)` will just insert the data with new timestamp and the latest time will be noted, better to make sure it is not replaced in the query?